libs/community/langchain_community/chat_models/llamacpp.py

```
@@ -362,6 +363,8 @@ def bind_tools(
                         f"Tool choice {tool_choice=} was specified, but the only "
                         f"provided tools were {tool_names}."
                     )
+            elif isinstance(tool_choice, str) and tool_choice in ['any', 'auto']:
+                tool_choice = formatted_tools[0]
```

IMO, if `tools` is length one, then we should translate `"any"` as you do here, but if there are multiple tools, we should raise an error.